packages/nextjs/src/server/authenticateRequest.ts

```diff
@@ -0,0 +1,70 @@
+import { constants } from '@clerk/backend';
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+
+import type { RequestState } from './clerkClient';
+import {
+  API_KEY,
+  API_URL,
+  clerkClient,
+  debugRequestState,
+  FRONTEND_API,
+  JS_VERSION,
+  PUBLISHABLE_KEY,
+  SECRET_KEY,
+} from './clerkClient';
+import type { WithAuthOptions } from './types';
+import { getCookie } from './utils';
+import { decorateResponseWithObservabilityHeaders } from './withClerkMiddleware';
+
+export const authenticateRequest = async (req: NextRequest, opts: WithAuthOptions) => {
+  const cookieToken = getCookie(req, constants.Cookies.Session);
+  const headers = req.headers;
+  const headerToken = headers.get('authorization')?.replace('Bearer ', '');
+  return await clerkClient.authenticateRequest({
+    ...opts,
+    apiKey: opts.apiKey || API_KEY,
+    secretKey: opts.secretKey || SECRET_KEY,
+    frontendApi: opts.frontendApi || FRONTEND_API,
+    publishableKey: opts.publishableKey || PUBLISHABLE_KEY,
+    cookieToken,
+    headerToken,
+    clientUat: getCookie(req, constants.Cookies.ClientUat),
+    origin: headers.get('origin') || undefined,
+    host: headers.get('host') as string,
+    forwardedPort: headers.get('x-forwarded-port') || undefined,
+    forwardedHost: headers.get('x-forwarded-host') || undefined,
+    referrer: headers.get('referer') || undefined,
+    userAgent: headers.get('user-agent') || undefined,
+    searchParams: new URL(req.url).searchParams,
+    // proxyUrl,
+    // isSatellite,
+    // domain,
+    // signInUrl,
+  } as any);
+};
+
+export const handleUnknownState = (requestState: RequestState) => {
+  const response = new NextResponse(null, { status: 401, headers: { 'Content-Type': 'text/html' } });
+  decorateResponseWithObservabilityHeaders(response, requestState);
+  return response;
+};
+
+export const handleInterstitialState = (requestState: RequestState, opts: WithAuthOptions) => {
+  const response = NextResponse.rewrite(
+    clerkClient.remotePublicInterstitialUrl({
```

Comment:
❓ Is it possible to throw the local interstitial in the new version? We should always prefer it if the framework allows it.

Reply:
I just reused the same helper `withClerkMiddleware` uses, nothing new - this shows as a new file because I copy-pasted it into a new file.

Shall we switch to the local insterstitial for both? This makes sense to me

Reply:
If next.js allows it for both, yes.

Reply:
As discussed, we will use local for the authMiddleware but continue using remote for withClerkMiddleware as in theory, its possible someone still uses a nextjs version that does not allow content responses from within middleware.